ts/features/wallet/onboarding/bancomat/screens/SearchBankScreen.tsx

```diff
@@ -1,18 +1,137 @@
-import { View } from "native-base";
+import { Content, Input, Item, Label, Text, View } from "native-base";
 import * as React from "react";
+import { FlatList, ListRenderItemInfo, SafeAreaView } from "react-native";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
+import { Link } from "../../../../../components/core/typography/Link";
+import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
 import { GlobalState } from "../../../../../store/reducers/types";
+import IconFont from "../../../../../components/ui/IconFont";
+import { BankPreviewItem } from "../components/BankPreviewItems";
+import { abis } from "../mock/mockData";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import I18n from "../../../../../i18n";
+import { navigateBack } from "../../../../../store/actions/navigation";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import { withLightModalContext } from "../../../../../components/helpers/withLightModalContext";
+import { LightModalContextInterface } from "../../../../../components/ui/LightModal";
+import TosBonusComponent from "../../../../bonus/bonusVacanze/components/TosBonusComponent";
 
+type Props = LightModalContextInterface &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
 /**
  * This screen allows the user to choose a specific bank to search for their Bancomat.
  * the user can also choose not to specify any bank and search for all Bancomat in his name
  * @constructor
  */
-export const SearchBankScreen: React.FunctionComponent = () => <View />;
+const SearchBankScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchText, setSearchText] = React.useState("");
+  const [filteredList, setFilteredList] = React.useState(
+    [] as typeof props.bankList
+  );
 
-const mapDispatchToProps = (_: Dispatch) => ({});
+  const [isSearchStarted, setIsSearchStarted] = React.useState(false);
 
-const mapStateToProps = (_: GlobalState) => ({});
+  const renderListItem = (isList: boolean) => (
+    info: ListRenderItemInfo<any>
+  ) => (
+    <BankPreviewItem
+      bank={info.item}
+      inList={isList}
+      onPress={props.onItemPress}
+    />
+  );
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchBankScreen);
+  const handleFilter = (text: string) => {
+    setIsSearchStarted(true);
```

Comment:
I don't know if when the user insert a query and after that cancel all the text and the `text.length===0` we should return to the previous visualization (`setIsSearchStarted(false);`) or not. What you think?

Reply:
This is my vote

```typescript
setSearchText(text);
 if(text.length === 0){
      setIsSearchStarted(false);
      setFilteredList([]);
    }
```

Reply:
yep, i was thinking for the same solution, in case a rework in a design review won't be a big trouble